SetupDataPkg/Library/SvdXmlSettingSchemaSupportLib/SvdXmlSettingSchemaSupport.c

```diff
@@ -0,0 +1,419 @@
+  <Settings />
+</ResultsPacket>
+
+**/
+XmlNode *
+EFIAPI
+New_ResultPacketNodeList (
+  EFI_TIME  *Date
+  )
+{
+  EFI_STATUS  Status;
+  XmlNode     *Root = NULL;
+  XmlNode     *Temp = NULL;
+  CHAR8       DateString[DATE_STRING_SIZE];
+
+  Status = CreateXmlTree (RESULT_XML_TEMPLATE, sizeof (RESULT_XML_TEMPLATE) - 1, &Root);
+  if (EFI_ERROR (Status)) {
+    DEBUG ((DEBUG_ERROR, "%a - Failed.  Status %r\n", __FUNCTION__, Status));
+    goto ERROR;
+  }
+
+  // Add AppliedOn Node
+  AsciiSPrint (&DateString[0], DATE_STRING_SIZE, "%d-%02d-%02dT%02d:%02d:%02d", Date->Year, Date->Month, Date->Day, Date->Hour, Date->Minute, Date->Second);
+  Status = AddNode (Root, RESULTS_APPLIED_ON_ELEMENT_NAME, &DateString[0], &Temp);
+  if (EFI_ERROR (Status)) {
+    DEBUG ((DEBUG_ERROR, "%a - Failed to add node for applied date. %r\n", __FUNCTION__, Status));
+    goto ERROR;
+  }
+
+  // Add SettingsList Node
+  Status = AddNode (Root, RESULTS_SETTINGS_LIST_ELEMENT_NAME, NULL, &Temp);
+  if (EFI_ERROR (Status)) {
+    DEBUG ((DEBUG_ERROR, "%a - Failed to add node for Settings. %r\n", __FUNCTION__, Status));
+    goto ERROR;
+  }
+
+  // Return the root
+  return Root;
+
+ERROR:
+  if (Root) {
+    FreeXmlTree (&Root);
+    // root will be NULL and null returned
+  }
+
+  return Root;
+}
+
+/**
+Function to get the SettingsPacket node given the root node
+
+**/
+XmlNode *
+EFIAPI
+GetSettingsPacketNode (
+  IN CONST XmlNode  *RootNode
+  )
+{
+  if (RootNode == NULL) {
+    DEBUG ((DEBUG_ERROR, "%a - RootNode is NULL\n", __FUNCTION__));
+    return NULL;
+  }
+
+  if (RootNode->XmlDeclaration.Declaration == NULL) {
+    DEBUG ((DEBUG_ERROR, "%a - RootNode is not the root node\n", __FUNCTION__));
+    ASSERT (RootNode->XmlDeclaration.Declaration != NULL);
+    return NULL;
+  }
+
+  if (AsciiStrnCmp (RootNode->Name, SETTINGS_PACKET_ELEMENT_NAME, sizeof (SETTINGS_PACKET_ELEMENT_NAME)) != 0) {
+    DEBUG ((DEBUG_ERROR, "%a - RootNode is not Settings Packet Element\n", __FUNCTION__));
+    return NULL;
+  }
+
+  return (XmlNode *)RootNode;
+}
+
+XmlNode *
+EFIAPI
+GetResultsPacketNode (
+  IN CONST XmlNode  *RootNode
+  )
+{
+  if (RootNode == NULL) {
+    DEBUG ((DEBUG_ERROR, "%a - RootNode is NULL\n", __FUNCTION__));
+    return NULL;
+  }
+
+  if (RootNode->XmlDeclaration.Declaration == NULL) {
+    DEBUG ((DEBUG_ERROR, "%a - RootNode is not the root node\n", __FUNCTION__));
+    ASSERT (RootNode->XmlDeclaration.Declaration != NULL);
+    return NULL;
+  }
+
+  if (AsciiStrnCmp (RootNode->Name, RESULTS_PACKET_ELEMENT_NAME, sizeof (RESULTS_PACKET_ELEMENT_NAME)) != 0) {
+    DEBUG ((DEBUG_ERROR, "%a - RootNode is not Result Settings Packet Element\n", __FUNCTION__));
+    return NULL;
+  }
+
+  // From root node there should only be 1 sibling which is the Packet Node
+  return (XmlNode *)RootNode;
+}
+
+XmlNode *
+EFIAPI
+GetSettingsListNodeFromPacketNode (
+  IN CONST XmlNode  *PacketNode
+  )
+{
+  ASSERT (PacketNode != NULL);

```

Comment:
CodeQL might flag that we only assert, not check for NULL? This is a copy/paste, but since we are changing it, perhaps worthwhile to change any of these assert only cases to check for NULL, assert, then return some error.

Reply:
Good call :) Updated in the latest update.

Reply:
Updated.